pkg/utils/common/controller.go

```diff
@@ -245,6 +245,22 @@ func (h *SuiteController) GetOpenshiftRoute(routeName string, routeNamespace str
 	return route, nil
 }
 
+// GetOpenshiftRouteByComponentName returns a route associated with the given component
+// Routes that belong to a given component will have the following label: 'app.kubernetes.io/name: <component-name>'
+func (h *SuiteController) GetOpenshiftRouteByComponentName(componentName string, componentNamespace string) (*routev1.Route, error) {
+	listOptions := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app.kubernetes.io/name=%s", componentName),
+	}
+	routeList, err := h.CustomClient.RouteClient().RouteV1().Routes(componentNamespace).List(context.Background(), listOptions)
+	if err != nil {
+		return &routev1.Route{}, err
+	}
+	if len(routeList.Items) == 0 {
+		return &routev1.Route{}, fmt.Errorf("unable to find routes with label %v:%v", "app.kubernetes.io/name", componentName)
+	}
+	return &routeList.Items[0], nil
```

Comment:
out of curiosity, why are we returning the first item from this list? is it because this list will only create 1 element always?

Reply:
The component name is unique, so the route should be unique as well

Reply:
I see, so the `routeList` will always contain 1 route (if there are no errors ofc). Makes sense.